### PR TITLE
Drop support for buggy and problematic alternative Python build method

### PR DESCRIPTION
**Pull request by pmatilai**
This was a kind of a necessary evil to see through us the Python 3 transition but with *that* over, there's little justification for maintaining an alternative build method to accumulate bugs and deprecated cruft on.

The one useful thing it provides is egg-info, but that's simple enough that we don't need Python libraries to do it, just generate it from the cmake build.
    
Fixes: #130, #2230

### CODE REVIEW

**Review by Conan-Kudo — APPROVED** [review]

**Comment by pmatilai** [comment]
The cmake build has ways to go before we can start actually recommending it (the inability to run the test-suite being the biggie of course), but the sooner people start testing it on their distros and systems etc, the better.

**Comment by junaruga** [comment]
I am using the `setup.py.in` in the `rpm-py-installer` pip package that installs the RPM Python binding pip package, and enables people to use the RPM python binding on some Python environments.
https://github.com/junaruga/rpm-py-installer

**Comment by pmatilai** [comment]
Yep, tough cookies. It simply doesn't make sense for us to try and support two things to achieve the same thing, and poorly at that.

**Comment by junaruga** [comment]
OK. Is there a way to create a PIP package from the `*.egg-info` file (`python/rpm.egg-info.in`)?

File: python/rpm.egg-info.in [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+Name: @PROJECT_NAME@
+Version: @PROJECT_VERSION@
+Summary: Python bindings for rpm
+Home-page: @PROJECT_HOMEPAGE_URL@
+Author: Rpm community
+Author-email: rpm-maint@lists.rpm.org
+License: GNU General Public License v2
+Description: Python bindings for rpm
+Platform: UNKNOWN
+
```
Comment (junaruga) [inline python/rpm.egg-info.in:11]: This blank line is intentional?

Comment (pmatilai) [reply python/rpm.egg-info.in]: Nope, although it's hardly consequential either. Fixed, shrug.

**Comment by pmatilai** [comment]
I know nothing about creating PIP packages, and a PR is hardly the place to discuss it anyway.

**Pull request merged**